Modules/_struct.c

```diff
@@ -2164,6 +2164,19 @@ s_sizeof(PyStructObject *self, void *unused)
     return PyLong_FromSize_t(size);
 }
 
+static PyObject *
+s_repr(PyStructObject *self)
+{
+    PyObject* fmt;
+    PyObject* s;
+
+    fmt = PyUnicode_FromStringAndSize(PyBytes_AS_STRING(self->s_format),
```

Comment:
Missing a check for this function returning NULL.

Also, this feels somewhat roundabout, wonder if there's a way to do this without going through an intermediate Unicode object.

Reply:
`s_get_format` function which implements `s.format` getter is doing the same thing without error checking.
Should I fix them both or leave it as is?

Reply:
That one is fine as it immediately returns the value from `PyUnicode_FromStringAndSize`, so the exception will get propagated.